frontend/views/components/ContributionItem.vue

```diff
@@ -0,0 +1,160 @@
+  props: {
+    who: [String, Array],
+    what: String,
+    action: {
+      type: String,
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['RECEIVING', 'GIVING'].indexOf(value) !== -1
+      },
+      default: 'RECEIVING'
+    },
+    type: {
+      type: String,
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['NON_MONETARY', 'MONETARY'].indexOf(value) !== -1
+      },
+      default: 'NON_MONETARY'
+    }
+  },
+  data () {
+    return {
+      isVisible: false
+    }
+  },
+  computed: {
+    contributionText () {
+      if (this.hasWhoElse) {
+        const html = {
+          0: `<span class="has-text-bold">${this.what}</span>`,
+          1: `<button class="is-unstyled is-link-inherit link">${this.otherContributor}`,
+          2: '</button>'
+        }
+
+        if (this.action === 'RECEIVING' && this.type === 'MONETARY') {
+          return L('{0} from {1} members{2}', html)
+        } else {
+          return L('A total of {0} to {1} members{2}', html)
+        }
+      } else {
+        if (this.action === 'RECEIVING') {
+          const html = {
+            0: `<span class="has-text-bold">${this.what}</span>`,
+            1: this.firstWho
+          }
+          return L('{0} by {1}', html)
+        } else {
+          if (this.type === 'MONETARY') {
+            const html = {
+              0: `<span class="has-text-bold">${this.what}</span>`,
+              1: this.firstWho
+            }
+            return L('{0} to {1}', html)
+          } else {
+            return `<span class="has-text-bold">${this.what}</span>`
+          }
+        }
+      }
+    },
+
+    listOfName () {
+      const html = {
+        0: `<span class="has-text-bold">${this.what}</span>`,
+        1: this.who.map((name, index) => {
+          return `<p class="has-text-1 c-contribution-list-item">${name}</p>`
+        }).join('')
+      }
+      if (this.action === 'RECEIVING') {
+        return L('{0} from {1}', html)
+      } else {
+        return L('A total of {0} to {1}', html)
+      }
+    },
+
+    firstWho () {
+      const who = this.who
+      if (!Array.isArray(who)) return who
+      return who.length === 2 ? this.L('{who0} and {who1}', { who0: who[0], who1: who[1] }) : who[0]
+    },
+
+    otherContributor () {
+      return this.who.slice(1).length
+    },
+
+    hasWhoElse () {
+      return Array.isArray(this.who) && this.who.length > 2
+    },
+    iconClass () {
+      const style = {
+        'NON_MONETARY': {
+          icon: 'heart',
+          color: 'warning'
+        },
+        'MONETARY': {
+          icon: 'dollar-sign',
+          color: 'success'
+        }
+      }
+      return `icon-${style[this.type].icon} c-icon-round has-background-${style[this.type].color} has-text-${style[this.type].color}`
+    }
+  }
+}
+</script>
+
+<style lang="scss">
+@import "../../assets/style/_variables.scss";
```

Comment:
For consistency throughout the project, use the new `@import "@assets/style/_variables.scss";` syntax